src/map.rs

```diff
@@ -234,6 +234,32 @@ impl Map<String, Value> {
         }
     }
 
+    #[cfg(all(feature = "preserve_order", not(no_btreemap_retain)))]
+    /// Retains only the elements specified by the predicate.
+    ///
+    /// In other words, remove all pairs `(k, v)` such that `f(&k, &mut v)` returns `false`.
+    /// The elements are visited in ascending key order.
+    #[inline]
+    pub fn retain<F>(&mut self, f: F)
+    where
+        F: FnMut(&String, &mut Value) -> bool,
+    {
+        self.map.retain(f);
+    }
+
+    #[cfg(all(not(feature = "preserve_order"), not(no_btreemap_retain)))]
+    /// Retains only the elements specified by the predicate.
+    ///
+    /// In other words, remove all pairs `(k, v)` such that `f(&k, &mut v)` returns `false`.
+    /// The elements are visited in ascending key order.
+    #[inline]
+    pub fn retain<F>(&mut self, f: F)
+    where
+        F: FnMut(&String, &mut Value) -> bool,
+    {
+        self.map.retain(f);
+    }
```

Comment:
Why are there two identical copies of this function? (Maybe I'm not observant enough to notice the difference &mdash; what is the difference between these?)

Reply:
Thanks for catching my mistake @dtolnay 

They were originally different but eventually became the same. Combination of late night coding and distraction of kids. It should be good to go now.